packages/jest-runner/src/run_test.js

```diff
@@ -106,22 +114,49 @@ export default (async function runTest(
     );
     const testCount =
       result.numPassingTests + result.numFailingTests + result.numPendingTests;
+
+    result.leaks = environmentLeakDetector;
     result.perfStats = {end: Date.now(), start};
     result.testFilePath = path;
     result.coverage = runtime.getAllCoverageInfo();
     result.sourceMaps = runtime.getSourceMapInfo();
     result.console = testConsole.getBuffer();
     result.skipped = testCount === result.numPendingTests;
     result.displayName = config.displayName;
+
     if (globalConfig.logHeapUsage) {
       if (global.gc) {
         global.gc();
       }
       result.memoryUsage = process.memoryUsage().heapUsed;
     }
+
     // Delay the resolution to allow log messages to be output.
-    return new Promise(resolve => setImmediate(() => resolve(result)));
+    await new Promise(resolve => setImmediate(resolve));
+
+    return result;
```

Comment:
Is this just simplifying the code or is there another reason you made this change?